File: sdk/python/examples/claude/01_hello_world.py
# Copyright (c) 2025 Agentspan
# Licensed under the MIT License. See LICENSE file in the project root for details.

"""Hello World — simplest Claude Code Agent with read-only tools.

Demonstrates:
    - Creating a ClaudeCodeAgent (Tier 1 passthrough)
    - Running it with AgentRuntime
    - Session durability: state is checkpointed to Agentspan server after each tool call

Requirements:
    - AGENTSPAN_SERVER_URL=http://localhost:8080/api
    - claude CLI installed and authenticated (`claude --version`)
    - claude-agent-sdk Python package installed
"""

from agentspan.agents.frameworks import ClaudeCodeAgent
from agentspan.agents import AgentRuntime

agent = ClaudeCodeAgent(
    name="hello_agent",
    allowed_tools=["Read", "Glob", "Grep"],
    cwd=".",
    max_turns=10,
)

if __name__ == "__main__":
    with AgentRuntime() as runtime:
        result = runtime.run(agent, "List the Python files in the current directory and summarize what this project does.")
        print(f"Status: {result.status}")
        result.print_result()

File: sdk/typescript/examples/langgraph/01-hello-world.ts
/**
 * Hello World -- simplest LangGraph agent with no tools.
 *
 * Demonstrates:
 *   - Using createReactAgent from @langchain/langgraph/prebuilt
 *   - Running a graph via Agentspan runtime.run() passthrough
 */

import { createReactAgent } from '@langchain/langgraph/prebuilt';
import { ChatOpenAI } from '@langchain/openai';
import { AgentRuntime } from '../../src/index.js';

// ---------------------------------------------------------------------------
// Build the graph
// ---------------------------------------------------------------------------
const llm = new ChatOpenAI({ model: 'gpt-4o-mini', temperature: 0 });
const graph = createReactAgent({ llm, tools: [] });

const PROMPT = 'Say hello and tell me a fun fact about Python programming.';

// ---------------------------------------------------------------------------
// Run on agentspan
// ---------------------------------------------------------------------------
async function main() {
  const runtime = new AgentRuntime();
  try {
    const result = await runtime.run(graph, PROMPT);
    console.log('Status:', result.status);
    result.printResult();
  } finally {
    await runtime.shutdown();
  }
}

main().catch(console.error);
